tests/e2e/e2e_trafficsplit_with_zero_weight_test.go

```diff
@@ -0,0 +1,255 @@
+package e2e
```

Comment:
Instead of an e2e test just for this, can we simply add a unit test or a scenario test? An e2e test seems too heavy for this sort of testing and a unit test or test under `tests/scenarios` would be much more efficient.

Reply:
Based on an offline discussion with Shashank I have removed the e2e test.  I have not added a unit test as this code path is likely to change with the routes refactor.